release-25.3: server: only auto-upgrade system tenant from meta1 leaseholder

stevendanna opened the pull request:
Backport 1/1 commits from #153464 on behalf of @stevendanna.

----

Running the auto-upgrade process from all nodes does some amount of wasted work. Let's avoid that by only running it from the meta1 leaseholder. This only applies to the system tenant.

For shared-process multi-tenancy, we may consider allowing secondary tenants access to this same information.

For separate process multi-tenancy, a different solution would be needed.

Informs #152916

Release note: None

----

Release justification: Default-off mitigation for potential source of cluster instability during auto-upgrade.

stevendanna committed:
server: only auto-upgrade system tenant from meta1 leaseholder

Running the auto-upgrade process from all nodes does some amount of
wasted work. Let's avoid that by only running it from the meta1
leaseholder. This only applies to the system tenant.

For shared-process multi-tenancy, we may consider allowing secondary
tenants access to this same information.

For separate process multi-tenancy, a different solution would be needed.

Informs #152916

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154355)
<!-- Reviewable:end -->

stevendanna committed:
server: check for isMeta1Leaseholder later in loop

This checks for isMeta1Leaseholder after first checking the version,
allowing the version checking goroutine to exit early for the common
case of being on the most recent version.

Epic: none
Release note: None
